plugins/bkpr/bookkeeper.c

```diff
@@ -414,58 +472,80 @@ static struct command_result *json_list_account_events(struct command *cmd,
 
 	res = jsonrpc_stream_success(cmd);
 	json_array_start(res, "events");
-	for (size_t i = 0, j = 0, k = 0;
-	     i < tal_count(chain_events)
-	     || j < tal_count(channel_events)
-	     || k < tal_count(onchain_fees);
-	     /* Incrementing happens inside loop */) {
-		struct channel_event *chan;
-		struct chain_event *chain;
-		struct onchain_fee *fee;
-		u64 lowest = 0;
+	json_add_events(res, channel_events, chain_events, onchain_fees);
+	json_array_end(res);
+	return command_finished(cmd, res);
+}
 
-		if (i < tal_count(chain_events))
-			chain = chain_events[i];
-		else
-			chain = NULL;
-		if (j < tal_count(channel_events))
-			chan = channel_events[j];
-		else
-			chan = NULL;
-		if (k < tal_count(onchain_fees))
-			fee = onchain_fees[k];
-		else
-			fee = NULL;
+static struct command_result *param_outpoint(struct command *cmd,
+					     const char *name,
+					     const char *buffer,
+					     const jsmntok_t *tok,
+					     struct bitcoin_outpoint **outp)
+{
+	*outp = tal(cmd, struct bitcoin_outpoint);
+	if (json_to_outpoint(buffer, tok, *outp))
+		return NULL;
+	return command_fail_badparam(cmd, name, buffer, tok,
+				     "should be a txid:outnum");
+}
 
-		if (chain)
-			lowest = chain->timestamp;
+static struct command_result *json_edit_desc_utxo(struct command *cmd,
+						  const char *buf,
+						  const jsmntok_t *params)
+{
+	struct json_stream *res;
+	struct bitcoin_outpoint *outpoint;
+	const char *new_desc;
+	struct chain_event **chain_events;
 
-		if (chan
-		    && (lowest == 0 || lowest > chan->timestamp))
-			lowest = chan->timestamp;
+	if (!param(cmd, buf, params,
+		   p_req("identifier", param_outpoint, &outpoint),
+		   p_req("description", param_string, &new_desc),
+		   NULL))
+		return command_param_failed();
 
-		if (fee
-		    && (lowest == 0 || lowest > fee->timestamp))
-			lowest = fee->timestamp;
+	db_begin_transaction(db);
+	edit_utxo_description(db, outpoint, new_desc);
+	chain_events = get_chain_events_by_outpoint(cmd, db, outpoint, true);
```

Comment:
Is returning the events with updated descriptions meant as confirmation that the command was successful and rows were updated? How about failing the command with an error (eg `no events found for description`) if no rows were updated? Would make it a lot easier to handle error states when using the command via RPC or gRPC. If there is a need to show the results, querying events for an id is easy enough with #7536

Reply:
Yes, returning the results is a convenience in case you need/want to display the result to a user in an interface.

I go back on forth on whether or not to error on "there's nothing to update that matches that description". Returning an empty result set seems like a fine way of signaling that no results were found -- nothing got updated? Really a matter of taste, and I just so happen to dislike "noisy errors" -> nothing wrong happened during execution it just so happens that the input you put in caused no effect.

Reply:
>  I just so happen to dislike "noisy errors" -> nothing wrong happened during execution it just so happens that the input you put in caused no effect

The explicit intention of the command is that *something happens* so imho nothing happening is an error. But returning an empty result is probably more in line with other CLN commands. As a caller you just have to be careful to check for the length of returned results > 0 and handle that accordingly.